runtime/bindings/python/iree/runtime/array_interop.py

```diff
@@ -129,6 +136,35 @@ def _map_to_host(self) -> Tuple[MappedMemory, np.ndarray]:
             host_array = host_array.astype(self._override_dtype)
         return mapped_memory, host_array
 
+    def _copy_to_host(self) -> np.ndarray:
+        # TODO: When synchronization is enabled, need to block here.
+        raw_dtype = self._get_raw_dtype()
+        allocation_size = np.prod(self._buffer_view.shape) * raw_dtype.itemsize
+        host_buffer = self._device.allocator.allocate_buffer(
+            memory_type=(MemoryType.HOST_LOCAL | MemoryType.DEVICE_VISIBLE),
+            allowed_usage=(BufferUsage.TRANSFER_TARGET | BufferUsage.MAPPING_SCOPED),
+            allocation_size=allocation_size,
+        )
+        self._cb.copy(self._buffer_view.get_buffer(), host_buffer, end=True)
+        # Synchronize and wait for copy to be complete.
+        sem = self._device.create_semaphore(0)
+        self._device.queue_execute(
+            [self._cb],
+            wait_semaphores=HalFence.create_at(sem, 0),
+            signal_semaphores=HalFence.create_at(sem, 1),
+        )
```

Comment:
since you are doing a single copy here you can use iree_hal_device_queue_copy to avoid needing to manage the command buffer

Reply:
Done!